### ISSUE TITLE
Is the lesson pacing different between tsurukame and the WaniKani web app?

### ISSUE THREAD

**Issue by slimemold** [davidsansome/tsurukame#697]
I noticed something interesting with the pacing between the tsurukame app (running on iOS 17) and the plain WaniKani web app.

When I go to the WaniKani web app, I'm seeing "Today's Lessons Done!" and "Reviews 0". However, on the tsurukame app, I'm showing 20 lessons (and Reviews 0, which matches the web app).

Should the available lessons match between tsurukame and the WaniKani web app? Note that I pretty much just started (currently level 2), and so I have not changed settings on either, so everything is at their defaults.

**Comment by UInt2048** [comment]
Tsurukame shows all available lessons by default (old behavior) with the understanding you'll pace yourself.

WaniKani introduced a lesson pacing thing on the site to encourage you not to do too many lessons per day but they don't actually stop you from doing that.

**Comment by HealsCodes** [comment]
Commenting because I experienced something similar today and it turns out the extra lessons that Tsurukame shows are hidden behind an "[🪄 Advanced]" button in my Lessons box on WaniKani - if Tsurukame can detect that it would be nice to have an option to not show them straight away.

**Comment by UInt2048** [comment]
The number of lessons is configurable on WaniKani: https://tofugu.notion.site/User-Setting-for-Number-of-Today-s-Lessons-b32c4bae5cf44aa6baa860711eff4a91

Implementing it here would be a little difficult, because we'd need to keep track of a Heatmap-like awareness of how many lessons you've done today online and offline, which we currently don't.

**Comment by HealsCodes** [comment]
> The number of lessons is configurable on WaniKani: https://tofugu.notion.site/User-Setting-for-Number-of-Today-s-Lessons-b32c4bae5cf44aa6baa860711eff4a91
> 
> 
> 
> Implementing it here would be a little difficult, because we'd need to keep track of a Heatmap-like awareness of how many lessons you've done today online and offline, which we currently don't.

I guess what I was referring to is if the API allows to detect the difference between normal available lessons and these:

![image](https://github.com/davidsansome/tsurukame/assets/304352/016d290b-8bc6-4413-a14f-44ac45db9326)

**Comment by UInt2048** [comment]
@HealsCodes No, there's no functional difference in the API. All of them are available.

The site's "rate limiting" is purely cosmetic. You can achieve the same effect by just limiting yourself to not do more than that many lessons in a day, which is exactly the same behavior.

**Comment by albertof92** [comment]
> @HealsCodes No, there's no functional difference in the API. All of them are available.
> 
> The site's "rate limiting" is purely cosmetic. You can achieve the same effect by just limiting yourself to not do more than that many lessons in a day, which is exactly the same behavior.

Actually the problem is not the limiting part for me, but instead that Tsurukame groups everyting together depending on the order you chose. So you have either you do all the radicals, kanjis or vocabulary in a row (see the screenshot). I want to be able to mix things up as it happens in Wanikani, where you have usually 1 kanji, one radical and three vocabs per batch.
![IMG_9928](https://github.com/davidsansome/tsurukame/assets/75002298/5e9be04c-ee43-46f5-bea9-39b2d5827f1e)

![IMG_9929](https://github.com/davidsansome/tsurukame/assets/75002298/26294017-a57a-4a7b-9695-211390c3125e)

**Comment by HealsCodes** [comment]
> @HealsCodes No, there's no functional difference in the API. All of them are available.
> 
> The site's "rate limiting" is purely cosmetic. You can achieve the same effect by just limiting yourself to not do more than that many lessons in a day, which is exactly the same behavior.

I think that's the behaviour I'm seeing - the "Today's Lesson Limit" in WK clips the Lessons on I can see to "15 available". The "Apprentice Limit" in Tsurukame - which I confused this with - is something completely different. 

If I set a minimum "Apprentice Limit" of 25 it blocks me from doing any new lessons at all right now but still displays "52 available" which got me really confused. 

I guess I'll exclusively do Lessons on the web version for now and limit Tsurukame to Reviews :)

**Comment by UInt2048** [comment]
@HealsCodes The apprentice limit is based on how many Apprentice SRS stage items you have and has nothing to do with the other limit

@syndual I tried to do this in #396 but I couldn't even get #548 merged

**Comment by syndual** [comment]
> @HealsCodes The apprentice limit is based on how many Apprentice SRS stage items you have and has nothing to do with the other limit
> 
> @syndual I tried to do this in #396 but I couldn't even get #548 merged

@davidsansome plz if we don't introduce something like this many people need to go back and forth between the Tsurukame and the webapp... Thank you
